packages/components/src/navigator/navigator-screen/component.tsx

```diff
@@ -48,17 +45,26 @@ function UnconnectedNavigatorScreen(
 	props: Props,
 	forwardedRef: ForwardedRef< any >
 ) {
+	const screenId = useRef( uuid() );
```

Comment:
you're right, I initially thought it was not suited for it but since we're always initializing in the same component it works, we can also now start moving to React's new useId

Reply:
Good idea, I forgot that we now have access about `useId` .

It would be good to see if we can migrate `useInstanceId` to internally also use `useId`, or if we can deprecated it and migrated all usages to `useId` directly.